### PR TITLE
small tx-pool refactoring 

### PR DESCRIPTION
**Pull request by zhangsoledad**
proposal changes:
* rename ContextualTransactionVerifier -> TimeRelativeTransactionVerifier
* split NonContextualTransactionVerifier from TransactionVerifier
* check syntactic correctness first before 
* refactory tx-pool rejection error
* re-broadcast when duplicated tx submit

### CODE REVIEW

File: rpc/src/module/pool.rs [hunk 1]
```diff
@@ -100,20 +100,24 @@ impl PoolRpc for PoolRpcImpl {
...
-                    .entry(peer_index)
-                    .or_default()
-                    .insert(hash.clone());
-                Ok(hash.unpack())
+                let tx_hash = tx.hash();
+                broadcast(tx_hash.clone());
+                Ok(tx_hash.unpack())
             }
-            Err(e) => Err(RPCError::from_ckb_error(e)),
+            Err(e) => Err(RPCError::submit_transaction_error(e, broadcast)),
```
Comment (doitian) [inline rpc/src/module/pool.rs:120]: It's weird to pass `broadcast` to the error conversion function. I would suggest match the returned error and broadcast on the duplication error.

**Comment by doitian** [comment]
bors r=quake,doitian

**Comment by doitian** [comment]
bors r=quake,doitian

**Comment by doitian** [comment]
bors retry

**Comment by doitian** [comment]
bors retry

**Comment by zhangsoledad** [comment]
benchmark

**Comment by nervos-bot-user** [comment]
#### Benchmark Result
  - TPS: 344.98
  - Samples Count: 51
  - CKB Version: 27012052dcf6ea81d5da94469009832db1559216
  - Instance Type: c5.xlarge
  - Instances Count: 3
  - Bench Type: 2in2out
  - CKB Logger Filter: info,ckb=debug

**Comment by zhangsoledad** [comment]
bors r=quake,doitian

**Comment by zhangsoledad** [comment]
bors retry

**Pull request merged**